### PR TITLE
pkg/nimble/autoconn: add user event callback

### PR DESCRIPTION
**Pull request by haukepetersen**
### Contribution description
When working with `nimble_autoconn` it would sometimes be very useful as a user, to get notified on connection change events, e.g. to trigger certain high-level network protocol events on connection state changes. 

This PR adds the possibility to register a user callback with `nimble_autoconn`, that simply passes through the `nimble_netif` events.

The PR also contains some improvements on the DEBUG output for `autoconn`.

### Testing procedure
Run `gnrc_networking` for any `nrf52x` board using `USEMODULE=nimble_autoconn_ipsp`. Everything should work just as before.

When enabling the DEBUG output for `pkg/nimble/autoconn/nimble_autoconn.c`, you should see a little nicer output like `[autoconn] CONNECT master ([HANDLE]|[BLE ADDRESS])`, where handle and address are now printed for every event instead of just the `CONNECT` ones.

### Issues/PRs references
none

### CODE REVIEW

**Comment by haukepetersen** [comment]
rebased after #12984 was merged.

File: pkg/nimble/autoconn/nimble_autoconn.c [hunk 1]
```diff
@@ -297,8 +310,8 @@ int nimble_autoconn_update(const nimble_autoconn_params_t *params,
 
 void nimble_autoconn_enable(void)
 {
-    DEBUG("[autoconn] ACTIVE\n");
     if (nimble_netif_conn_count(NIMBLE_NETIF_UNUSED) > 0) {
+        DEBUG("[autoconn] ACTIVE\n");
```
Comment (miri64) [inline pkg/nimble/autoconn/nimble_autoconn.c]: Unrelated?

File: pkg/nimble/autoconn/nimble_autoconn.c [hunk 2]
```diff
@@ -309,8 +322,8 @@ void nimble_autoconn_enable(void)
 
 void nimble_autoconn_disable(void)
 {
-    DEBUG("[autoconn] DISABLED\n");
     if ((_state == STATE_ADV) || (_state == STATE_SCAN)) {
+        DEBUG("[autoconn] DISABLED\n");
```
Comment (miri64) [inline pkg/nimble/autoconn/nimble_autoconn.c]: Unrelated?

File: pkg/nimble/autoconn/nimble_autoconn.c (hunk 1 above)
Comment (haukepetersen) [reply pkg/nimble/autoconn/nimble_autoconn.c]: its an improvement, it was actually kind of broken before...

File: pkg/nimble/autoconn/nimble_autoconn.c (hunk 2 above)
Comment (haukepetersen) [reply pkg/nimble/autoconn/nimble_autoconn.c]: same

**Comment by haukepetersen** [comment]
Murdock is happy as well -> go.
